src/Service.php

```diff
@@ -52,14 +52,17 @@ private function fetchFlags(): void
                 $this->flags->put($flag, new Flag($flag, $data['treatment'], $data['config'] ?? ''));
```

Comment:
I think this violates a library design. A library will provide you with sensible exceptions and then the app using this library can decide whether to log, send email or throw the exception.